LastPush/CanvasGapsGenerator/src/main/kotlin/jetbrains/kotlin/course/last/push/Main.kt

```diff
@@ -69,18 +69,12 @@ fun repeatHorizontally(pattern: String, n: Int, patternWidth: Int): String {
         sb.append(currentRow.repeat(n))
         sb.append(newLineSymbol)
     }
-    return sb.toString()
+    return sb.toString().dropLastWhile { it.toString() == newLineSymbol }
 }
 
-fun dropTopFromLine(line: String, width: Int, patternHeight: Int, patternWidth: Int): String {
-    val nToDrop = if (patternHeight > 1) {
-        patternWidth * width + newLineSymbol.length
-    } else {
-        0
-    }
-    val newPattern = line.removeSuffix(newLineSymbol).drop(nToDrop)
-    return "$newPattern$newLineSymbol"
-}
+fun dropTopFromLine(line: String, width: Int, patternHeight: Int, patternWidth: Int): String = if (patternHeight > 1) {
```

Comment:
Have you checked if the new implementation correlates with the task description?